lib/galaxy/jobs/runners/util/job_script/DEFAULT_JOB_FILE_TEMPLATE.sh

```diff
@@ -27,13 +27,14 @@ _galaxy_setup_environment() {
 
 $integrity_injection
 $slots_statement
-export GALAXY_SLOTS
 export PYTHONWARNINGS="ignore"
 GALAXY_VIRTUAL_ENV="$galaxy_virtual_env"
 _GALAXY_VIRTUAL_ENV="$galaxy_virtual_env"
 PRESERVE_GALAXY_ENVIRONMENT="$preserve_python_environment"
 GALAXY_LIB="$galaxy_lib"
 _galaxy_setup_environment "$PRESERVE_GALAXY_ENVIRONMENT"
+export _GALAXY_JOB_HOME_DIR
```

Comment:
```suggestion
```

`$env_setup_commands` above should set `HOME` to `$_GALAXY_JOB_HOME_DIR`, so this shouldn't be needed.

Reply:
Can you give me a link? Can't find it.

Reply:
I cannot give you a simple link unfortunately, it's a complex process:

- `JobWrapper.prepare()` (in `lib/galaxy/jobs/__init__.py`) calls `tool_evaluator.build()` to set its `environment_variables` attribute
- `ToolEvaluator.build()` (in `lib/galaxy/tools/evaluation.py`) calls `self.__build_environment_variables()`
- `ToolEvaluator.__build_environment_variables()` (in `lib/galaxy/tools/evaluation.py`) executes:

  ```python
  home_dir = self.compute_environment.home_directory()
  if home_dir:
      environment_variable = dict(name="HOME", value='"%s"' % home_dir, raw=True)
      environment_variables.append(environment_variable)
  ...
  self.environment_variables = environment_variables
  ```
  Note that `self.compute_environment.home_directory()` is equal `$_GALAXY_JOB_HOME_DIR` for tools with a profile >= 18.01
- Later, `BaseJobRunner.get_job_file()` (in `lib/galaxy/jobs/runners/__init__.py` ) generates the `env_setup_commands` list; in particular, for each environment variable in `job_wrapper.environment_variables`, a shell export statement is appended to `env_setup_commands` (one of this being `HOME="$_GALAXY_JOB_HOME_DIR"; export HOME`)
- `get_job_file()` passes `env_setup_commands` as parameter to `job_script()`
- `job_script()` (in `lib/galaxy/jobs/runners/util/job_script/__init__.py`) replaces the `$env_setup_commands` template placeholder in `_galaxy_setup_environment()` (in `lib/galaxy/jobs/runners/util/job_script/DEFAULT_JOB_FILE_TEMPLATE.sh` ) with the stringified value of its `env_setup_commands` argument